unix_integration/src/daemon.rs

```diff
@@ -281,15 +281,15 @@ async fn handle_client(
                 debug!("pam authenticate init");
                 pam_state = PamState::Step(account_id.clone());
                 cachelayer
-                    .pam_account_authenticate_step(account_id.as_str(), None)
+                    .pam_account_authenticate_step(account_id.as_str(), None, None)
                     .await
                     .unwrap_or(ClientResponse::Error)
             }
-            ClientRequest::PamAuthenticateStep(cred) => {
+            ClientRequest::PamAuthenticateStep(cred, data) => {
                 debug!("pam authenticate step");
                 match &pam_state {
                     PamState::Step(account_id) => match cachelayer
-                        .pam_account_authenticate_step(account_id.as_str(), cred)
+                        .pam_account_authenticate_step(account_id.as_str(), cred, data)
```

Comment:
Need a way to see whant the result from _step is here so we can potentially clear the auth session on completion so that a client can do a sequence of pam auths if required without dropping/opening the unix socket.

Reply:
That's what I was getting at in unix_integration/src/daemon.rs:300. If we sent a pam_prompt request, we're not done. Anything else (some other client response), and we know the auth request has completed. Maybe not the most elegant, but I did handle this.